extensions/blocks/map/map.php

```diff
@@ -126,6 +144,8 @@ function jetpack_map_block_render_single_block_page() {
 
 	add_filter( 'jetpack_is_amp_request', '__return_false' );
 
+	jetpack_record_mapbox_load_event();
```

Comment:
I am not sure I understand why we call this here again. Am I missing something?

Reply:
This was slightly confusing to me too, and I hope I got it right (cc @jeffersonrabb).

The Map block's AMP implementation is peculiar, or at least it's peculiar to me with no familiarity with AMP.

When loaded normally (not in AMP), this file is a simple server-side rendering of the Map block, similar to any other.

But it also has a function hooked to the `wp` action (that runs very early in the execution) that, if the URL has two Map block specific query params, instead of rendering the site, it renders **only** the map specified by the query params.
You can test this yourself with an URL like this:
`https://www.example.org?map-block-post-id={ id of a post containing Map blocks }&map-block-counter={ number of the Map block to render }`

Now, when we load the block in AMP, instead of normally rendering the Map block, we render an AMP iframe using that kind of URL as source:
`<amp-iframe src="https://example.org?map-block-post-id=123&map-block-counter=1` />`
In other words, instead of directly loading the map via JS, we load it into an iframe (to make AMP happy, I guess).

Because of this kind of switcheroo (and some early returns that would affect the tracking accuracy), I personally feel more confident in tracking the Mapbox load _as late as possible_, which is, right next to `Jetpack_Gutenberg::load_assets_as_required( 'map' );`. 😄